xdsl/dialects/func.py

```diff
@@ -69,6 +69,51 @@ def from_region(name: str, input_types: List[Attribute],
         op = FuncOp.build(attributes=attributes, regions=[region])
         return op
 
+    def replace_argument_type(self, arg: int | BlockArgument,
+                              new_type: Attribute):
+        """
+        Replaces a function argument with a new one
+        """
+
+        if isinstance(arg, int):
+            if len(self.body.blocks[0].args) <= arg:
```

Comment:
You may want to add a check here for negative values (even if you want to keep negative values).